tests/framework-tests/preferences/js/SelfVoicingEnactorTests.js

```diff
@@ -0,0 +1,209 @@
+        components: {
+            speak: {
+                type: "fluid.tests.prefs.enactor.speakEnactor"
+            },
+            speakTester: {
+                type: "fluid.tests.speakTester"
+            }
+        }
+    });
+
+    fluid.defaults("fluid.tests.speakTester", {
+        gradeNames: ["fluid.test.testCaseHolder", "autoInit"],
+        testOptions: {
+            sampleText: "Reading sample text",
+            startStopFireRecord: {
+                onStart: 1,
+                onStop: 1,
+                onTextQueued: 1
+            }
+        },
+        modules: [{
+            name: "fluid.prefs.enactor.speakEnactor",
+            tests: [{
+                expect: 2,
+                name: "Start-Stop flow",
+                sequence: [{
+                    func: "{speak}.announce",
+                    args: ["{that}.options.testOptions.sampleText"]
+                }, {
+                    listener: "fluid.tests.speakTester.verifyRecords",
+                    args: ["{speak}", "{that}.options.testOptions.startStopFireRecord", ["{that}.options.testOptions.sampleText"]],
+                    spec: {priority: "last"},
+                    event: "{speak}.events.onStop"
+                }]
+            }]
+        }]
+    });
+
+    fluid.tests.speakTester.verifyRecords = function (that, expectedEvents, expectedText) {
+        jqUnit.assertDeepEq("The events should have fired correctly", expectedEvents, that.eventRecord);
+        jqUnit.assertDeepEq("The text to announce should have been queued correctly", expectedText, that.speakQueue);
+        that.clearRecords();
+    };
+
+    /*******************************************************************************
+     * Unit tests for fluid.prefs.enactor.selfVoicingEnactor
+     *******************************************************************************/
+
+    fluid.defaults("fluid.tests.prefs.enactor.selfVoicingEnactor", {
+        gradeNames: ["fluid.prefs.enactor.selfVoicingEnactor", "fluid.tests.prefs.enactor.speakEnactor", "autoInit"],
+        model: {
+            value: false
+        },
+        strings: {
+            welcomeMsg: ""
+        },
+        invokers: {
+            toggle: {
+                changePath: "value",
+                value: "{arguments}.0"
+            }
+        }
+    });
+
+    fluid.defaults("fluid.tests.selfVoicingTests", {
+        gradeNames: ["fluid.test.testEnvironment", "autoInit"],
+        components: {
+            selfVoicing: {
+                type: "fluid.tests.prefs.enactor.selfVoicingEnactor",
+                container: ".flc-selfVoicing"
+            },
+            selfVoicingTester: {
+                type: "fluid.tests.selfVoicingTester"
+            }
+        }
+    });
+
+    fluid.defaults("fluid.tests.selfVoicingTester", {
+        gradeNames: ["fluid.test.testCaseHolder", "autoInit"],
+        testOptions: {
+            // Because events are fired for each utterance
```

Comment:
We should improve the event API of TTS so that this can be tested properly. It should have an "onQueueEmpty" event that fires once all speech has stopped.